starknet/src/voting_strategies/evm_slot_value.cairo

```diff
@@ -47,14 +53,40 @@ mod EvmSlotValueVotingStrategy {
             )>::deserialize(ref params)
                 .unwrap();
 
-            // Get the balance of the voter at the given block timestamp
+            let mut user_params = user_params;
+            let (checkpoint_index, checkpoint_mpt_proof, exclusion_mpt_proof) = Serde::<(
+                u32, Span<Span<u64>>, Span<Span<u64>>
+            )>::deserialize(ref user_params)
+                .unwrap();
+
+            // Get the slot key for the final checkpoint
+            let slot_key = InternalImpl::get_mapping_slot_key(
+                voter.into(), slot_index, checkpoint_index
+            );
+
+            // Get the slot containing the final checkpoint
             // Migration to components planned ; disregard the `unsafe` keyword,
             // it is actually safe.
             let state = SingleSlotProof::unsafe_new_contract_state();
-            let balance = SingleSlotProof::InternalImpl::get_storage_slot(
-                @state, timestamp, evm_contract_address, slot_index, voter.into(), user_params
+            let checkpoint = SingleSlotProof::InternalImpl::get_storage_slot(
+                @state, timestamp, evm_contract_address, slot_key, checkpoint_mpt_proof
             );
-            balance
+            assert(checkpoint.is_non_zero(), 'Slot is zero');
+
+            // Verify the checkpoint is indeed the final checkpoint by checking the next slot is zero.
+            let state = SingleSlotProof::unsafe_new_contract_state();
```

Comment:
can we use the same `state` as before? since it's borrowed earlier I think we can?

Reply:
yes probably, will test then update